### PR TITLE
Added error tracking for sms errors

### PR DESCRIPTION
**Pull request by stefaniacardenas**

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-250-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://250-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Error/index.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 import style from './style.css'
+import { sendScreen } from '../../Tracker'
 
-const Error = ({error}) => {
-  const errorText = errors[error.name]
-  const errorType = error.type === 'error' ? 'error' : 'warning';
-  return (
-    <div className={style[`container-${errorType}`]}>
+class Error extends Component {
+  errorName = this.props.error.name
```
Comment (rfreitas) [inline src/components/Error/index.js]: getting the props at init value is not correct react

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-250-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://250-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Error/index.js [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
+  componentDidMount() {
+    const errorName = this.props.error.name.toLowerCase()
+    if (this.errorType() === 'error' && this.isSmsError()) {
+      sendScreen([errorName])
+    }
+  }
+
+  isSmsError = () => {
+    const errorName = this.props.error.name
+    return errorName === 'SMS_FAILED' || errorName === 'SMS_OVERUSE'
```
Comment (rfreitas) [inline src/components/Error/index.js]: why is this necessary?

Comment (stefaniacardenas) [reply src/components/Error/index.js]: We only want to send the screen to woopra if it's an sms related error

Comment (rfreitas) [reply src/components/Error/index.js]: what are the other cases?
And if that is the cases then it's better to have a component called SMSError screen that sends that error, rather than having this complexity here.

Comment (stefaniacardenas) [reply src/components/Error/index.js]: These are the other use cases:
```'INVALID_CAPTURE'
  'INVALID_TYPE'
  'UNSUPPORTED_FILE'
  'INVALID_SIZE'
  'NO_FACE_ERROR'
  'MULTIPLE_FACES_ERROR'
  'SERVER_ERROR'
  'GLARE_DETECTED'```
All of these errors are tracked on the uploader screen, the one that is currently displaying the wrong message on Woopra (it always tracks an error screen for uploader. I could track all the errors in the Error component and this would also fix the issue for the uploader component.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-250-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://250-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-250-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
actually @stefaniacardenas  i forgot to mention yesterday. I think it is better for the event name to be `screen_crossDevice_sms_failed` instead of just `screen_sms_failed` no, as logically it sits under `screen_crossDevice`

**Comment by rfreitas** [comment]
on @seewah comment, I agree as well, do you know how to append?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-250-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-250-pr-onfido-sdk-ui-onfido.surge.sh
